### ISSUE TITLE
role naming choices

### ISSUE THREAD

**Issue by eengstrom** [maxhoesel/ansible-collection-smallstep#21]
Is there a reason you chose the names of the roles to be `ca_server` and `step_client` ?  To parallel the packages Smallstep provides, would the names not be better named `step_ca` and `step_cli`, respectively?

**Comment by maxhoesel** [comment]
Good point! I created these role names in a bit of a rush to get some working code pushed, but in hindsight they aren't optimal - like a lot of this collection right now. I was busy with some other projects the past few months, but I plan on working on this collection again now that I have some free time.

I just had a look through the two roles again, and there's quite a few things that I'd like to change. My current plan is to create two new roles named `step_ca` and `step_cli` and go from there, leaving the old ones in place with a deprecation message for now.

**Comment by eengstrom** [comment]
So, I have a strong reason to keep using what you have now, but also see the same room for improvement you mentioned here and in #20.    I have some additions, including a module to actually _create_ certificates.  I'd be happy to help, both design/scope/implement the next major version.  Perhaps a completely different issue would help to track?  

Either way, if you can keep a `devel` or `feature` branch running that incorporates the redesign changes, I'd be happy to evaulate and test.

**Comment by maxhoesel** [comment]
I've pushed a new `0.3_rework` branch for this redesign/cleanup, along with the new `step_cli` role. That branch can serve as the base for all the improvements to come.

I've created separate issues for everything that I would like to address right now, including module naming, testing strategy and some other minor things.

As for the certificates module, that sounds like a good idea! I guess that would involve wrapping around `step (ca) certificate`? Feel free to create a new issue for it.
